### PR TITLE
Python news reader

### LINKED ISSUE #475: Python voice assistant to fetch and read news from NewsAPI
# Description
A python script which will fetch news from newsapi.com and will speak that news with headlines and description.

## Language

- [ ] Bash
- [ ] Go
- [ ] Javascript
- [ ] Powershell
- [x] Python
- [ ] Other (Please specify)

## Type of issue

- [x] Feature
- [ ] Bug

## Checklist:

- **For Feature**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| Python News Assistant | A python script which will fetch news from newsapi.com and will speak that news with headlines and description. Reading news and articles sometimes may too time consuming and boring .With the help of this python news reader you listen to top news while working on your other work.  | Python, Hacktoberfest2020, newsapi.com | @KmrAnish04 |

**Is your feature request related to a problem? Please describe.:**
Sometimes reading news articles or news papers may too time consuming and boring. So this issues I have solved by making python news reader with voice over feature. So now you can listen to your news while messing with other work which is very interesting and time saving.

**Describe the solution you'd like:**
A simple python program with voice over feature which can read news from newsapi.com.

### PR DESCRIPTION
**Pull request by KmrAnish04**
# Description

A python script which will fetch news from newsapi.com and will speak that news with headlines and description.

Fixes #475

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by KmrAnish04** [comment]
HI @HarshCasper , Thanks for assigning this issue to me and here is my PR for that issue. There is showing "All checks have failed", what does it mean? or I have made this PR in wrong way? . Let me know everything has done right or not. If all right, please merge this small contribution from my side. Thankful to you @HarshCasper .

**Comment by KmrAnish04** [comment]
Hi @seema1711 , Can look into this PR, Why this is showing "All checks have failed" ? 
Is something wrong in my contribution? . Please, let me know with solution.

**Comment by KmrAnish04** [comment]
Hi @HarshCasper , Please look into this PR's and let me if something is missing. If all right, then merge this small contribution from my side. Thanks.

File: Python/Python_news_reader/Python news reader.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+                if key == 'description':
+                    description = str(dics[key])
+                    print(dics[key])
+                    speaker("Description")
+                    time.sleep(1)
+                    speaker(description)
+
+if __name__ == "__main__":
+    func()
```
Comment (HarshCasper) [inline Python/Python_news_reader/Python news reader.py:44]: Kindly add a new line signifying the End of File

File: Python/Python_news_reader/Python news reader.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+***Visit newsapi.org for documentation and for your API key***
+'''
+
+import time
+from win32com.client import Dispatch
+import requests
+
+def func():
+    url = (
+        'http://newsapi.org/v2/top-headlines?country=in&category=business&apiKey=paste your API key here')  # Only paste your API key here, don't change the whole link.
```
Comment (HarshCasper) [inline Python/Python_news_reader/Python news reader.py:14]: Put the Comment above the Line of Code to improve readibility

File: Python/Python_news_reader/Python news reader.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                    speaker(title)
+
+                if key == 'description':
+                    description = str(dics[key])
+                    print(dics[key])
+                    speaker("Description")
+                    time.sleep(1)
+                    speaker(description)
+
+if __name__ == "__main__":
```
Comment (HarshCasper) [inline Python/Python_news_reader/Python news reader.py:43]: Add a `requirements.txt` for your Script

File: Python/Python_news_reader/Readme.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+This python program fetches the news from Indain sources and speaks or reads the news with voice over feature. 
+A python script which will fetch news from newsapi.com and will speak that news with headlines and description. 
+Reading news and articles sometimes may too time consuming and boring .With the help of this python news reader 
+you listen to top news while working on your other work.
+
+### -What is News API?
+
+News API is a simple HTTP REST API for searching and retrieving live articles from all over the web.
+Visit the newsapi.org for documentaion and your personal API key.
```
Comment (HarshCasper) [inline Python/Python_news_reader/Readme.md:11]: Add a brief note on how someone can run your Script for better accessibility?

File: Python/Python_news_reader/Python news reader.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+def func():
+    url = (
+        'http://newsapi.org/v2/top-headlines?country=in&category=business&apiKey=paste your API key here')  # Only paste your API key here, don't change the whole link.
+    # Example: ' http://newsapi.org/v2/top-headlines?country=in&category=business&apiKey=9895kn34nah45kan87ha12HG34Ng3s6 '
+
+    response = requests.get(url)
+    m = response.json()
+    print(m)
+    lst = m['articles']
```
Comment (HarshCasper) [inline Python/Python_news_reader/Python news reader.py:20]: The Variable names don't seem to follow any specification. Try running your Script through PyLint and make apt changes.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Kindly make the suggested changes

**Comment by KmrAnish04** [comment]
Hello @HarshCasper , Thanks for your response. I will definitely improve my files according to your expectations and to increase the efficiency and readability of code and documentations.  I am a newbie so just give me some time. Thanks for your cooperation.

**Comment by KmrAnish04** [comment]
Hi @HarshCasper , I closed this PR by mistake, I am sorry about that. And the changes you asked to do in the above files is almost done by me. So now should I create a new pull request for these above changes that you have asked or there another way of doing it. Let me know how can  and work is almost done according to your expectations.

**Comment by KmrAnish04** [comment]
> ## I have modified all the files and added some new files according to expectations and to improve the efficiency of the program.

### Changes done by me:

* Added a new line signifying the End of File
* Added a requirements.md file for this Script
* Added a brief note on how someone can run this Script
* Improved variable names for better understanding
* Added "Url.txt" file for better efficiency
* Improved readme.md file

#### Link to new PR #511 

@HarshCasper

**Pull request closed without merging**